src/LayerDicts.jl

```diff
@@ -66,37 +64,19 @@ end
 
 Base.length(ld::LayerDict) = length(keys(ld))
 
-if VERSION < v"0.7.0-DEV.5126"
-    function Base.start(ld::LayerDict)
-        ld_keys = keys(ld)
-        return (ld_keys, start(ld_keys))
-    end
+function Base.iterate(ld::LayerDict, state=(keys(ld),))
+    ld_keys = first(state)
+    key_state = Base.tail(state)
```

Comment:
```suggestion
    key_state = Base.tail(state)
    @assert length(keystate) == 0 || length(keystate) == 1
```

Is my suggestion above correct? if it is not, then then this code is really confusing.

Reply:
That seems reasonable to me, though I'm not super familiar with the new iteration protocol.